src/algos/binary_search/prework/exercise.clj

```diff
@@ -0,0 +1,42 @@
+(ns algos.binary_search.prework.exercise)
+
+;An implementation of recursive binary search:
+(defn binary-search-recursive
+  [coll item]
+  (loop [coll coll]
+    (cond
+      (empty? coll)         false
+      (< item (first coll)) false
+      (> item (last coll))  false
+      :else
+        (let [midpoint (quot (count coll) 2)
+              midpoint-el (get coll midpoint)]
+          (cond
+            (= item midpoint-el) true
+            (< item midpoint-el) (recur (subvec coll 0 midpoint))
+            (> item midpoint-el) (recur (subvec coll (inc midpoint))))))))
+
+;4. Looking back
+; unlike Python's slice, subvec is O(1)
+; There is a memory cost to this but because Clojure uses structural sharing and implements vectors as trees, when we
+; create a 'new' vector we copy *references* to the original vector rather than the values themselves which saves having
+; to copy the entire vector.
+; (src: https://www.infoq.com/articles/in-depth-look-clojure-collections/, CMD+F 'Clojure's persistent collections' + https://hackernoon.com/how-immutable-data-structures-e-g-immutable-js-are-optimized-using-structural-sharing-e4424a866d56)
+;
+; After further thought there's ~no~ reason I have to manipulate or pass around the vector at all, I can just use the indices.
+; I also remove the check for whether it's within the given range bc I don't know if that goes against the implementation of binary search(?)
+; Doesn't seem like it should and it prevents you from doing unnecessary computations but jic...
```

Comment:
Good question (I see what you meant in class now).  The asymptotic complexity is still good, i.e. O(log n), if you do three checks instead of one each time you recurse.

That being said, I think the range checks are technically not needed (except on the outermost call), but there's really no harm in having it.